packages/expo-dev-launcher/android/src/main/java/expo/modules/devlauncher/modules/DevLauncherInternalModule.kt

```diff
@@ -0,0 +1,243 @@
+          }
+          controller.loadApp(appUrl, projectUrl)
+        } catch (e: Exception) {
+          promise.reject("ERR_DEV_LAUNCHER_CANNOT_LOAD_APP", e.message, e)
+          return@launch
+        }
+        promise.resolve(null)
+      }
+    }
+
+    AsyncFunction("loadApp") { url: String, promise: Promise ->
+      controller.coroutineScope.launch {
+        try {
+          val parsedUrl = sanitizeUrlString(url)
+          controller.loadApp(parsedUrl)
+        } catch (e: Exception) {
+          promise.reject("ERR_DEV_LAUNCHER_CANNOT_LOAD_APP", e.message, e)
+          return@launch
+        }
+        promise.resolve(null)
+      }
+    }
+
+    AsyncFunction<List<Bundle?>>("getRecentlyOpenedApps") {
+      val apps: MutableList<Bundle> = ArrayList()
+
+      for (recentlyOpenedApp in controller.getRecentlyOpenedApps()) {
+        val app = Bundle()
+
+        app.putDouble("timestamp", recentlyOpenedApp.timestamp.toDouble())
+        app.putString("name", recentlyOpenedApp.name)
+        app.putString("url", recentlyOpenedApp.url)
+        app.putBoolean("isEASUpdate", recentlyOpenedApp.isEASUpdate == true)
+
+        if (recentlyOpenedApp.isEASUpdate == true) {
+          app.putString("updateMessage", recentlyOpenedApp.updateMessage)
+          app.putString("branchName", recentlyOpenedApp.branchName)
+        }
+
+
+        apps.add(app)
+      }
+      return@AsyncFunction apps
+    }
+
+    AsyncFunction("clearRecentlyOpenedApps") {
+      controller.clearRecentlyOpenedApps()
+    }
+
+    AsyncFunction("getPendingDeepLink") { promise: Promise ->
+      intentRegistry.intent?.data?.let {
+        promise.resolve(it.toString())
+      }
+
+      promise.resolve(intentRegistry.intent?.action)
+    }
+
+    AsyncFunction<WritableMap?>("getCrashReport") {
+      val registry = DevLauncherErrorRegistry(context)
+
+      return@AsyncFunction registry.consumeException()?.toWritableMap()
+    }
+
+    AsyncFunction("getBuildInfo") {
+      val map = Bundle()
+      val packageManager = context.packageManager
+      val packageName = context.packageName
+
+      val packageInfo = packageManager.getPackageInfo(packageName, 0)
+      val applicationInfo = packageManager.getApplicationInfo(packageName, PackageManager.GET_META_DATA)
+      val appName = packageManager.getApplicationLabel(applicationInfo).toString()
+      val runtimeVersion = controller.updatesInterface?.runtimeVersion
+      val appIcon = getApplicationIconUri()
+
+      val updatesUrl = controller.updatesInterface?.updateUrl
+      val appId = if (updatesUrl !== null) {
+        updatesUrl.lastPathSegment ?: ""
+      } else {
+        ""
+      }
+
+      map.apply {
+        putString("appVersion", packageInfo.versionName)
+        putString("appId", appId)
+        putString("appName", appName)
+        putString("appIcon", appIcon)
+        putString("runtimeVersion", runtimeVersion)
+      }
+
+      return@AsyncFunction map
```

Comment:
```suggestion
      return@AsyncFunction Bundle().apply {
        putString("appVersion", packageInfo.versionName)
        putString("appId", appId)
        putString("appName", appName)
        putString("appIcon", appIcon)
        putString("runtimeVersion", runtimeVersion)
      }
```
Probably no need to have a variable for this but up to you

Reply:
fixed